experimental/GModule/Cohomology.jl

```diff
@@ -402,20 +402,14 @@ Compute an fp-presentation of the group generated by 'g'
 and returns both the group and the map from the new group to the
 parent of the generators.
 """
-function fp_group(g::Vector{<:Oscar.GAPGroupElem})
+function fp_group_with_isomorphism(g::Vector{<:Oscar.GAPGroupElem})
```

Comment:
This is not an issue with this PR, but rather with the existing code (I pointed it out to @fieker before): this function does not actually use `g`, it just uses it to get `parent(g[1])`, checks that all elements of `g` have the same parent, and then ignores `g`. It also seems to always be called with `gens(G)` as argument; so I really wonder why this function doesn't just take a `GAPGroup` as argument to begin with.

Assuming such a change, a user may then wonder what difference between `fp_group_with_isomorphism(G)` and `isomorphism(FPGroup, G)` is... apparently just the use of `IsomorphismFpGroupByGenerators` versus `IsomorphismFpGroup`. This is an important difference of course, but perhaps we can and should do better when it comes to aligning this? E.g. there could be a kwarg `preserve_generators::Bool` which indicates whether we prefer an isomorphism that "imitates" the generating sequence of the source in the domain (`true`); or would prefer an isomorphism that's as efficient as possible (`false`).

Reply:
For completeness: another difference between `fp_group_with_isomorphism` and `isomorphism(FPGroup, ...)` is the direction of the map.